webapp/graphite/util.py

```diff
@@ -36,7 +36,18 @@
 if sys.version_info >= (3, 0):
   PY3 = True
   import pickle
-  from io import BytesIO as StringIO
+  from io import BytesIO
+
+
+  class StringIO(BytesIO):
+    def __init__(self, buffer=None):
+      if buffer is None:
+        super().__init__()
+      elif isinstance(buffer, six.string_types):
+        super().__init__(buffer.encode('utf-8'))
+      else:
+        super().__init__(buffer)
+
```

Comment:
@takluyver I came up with this as an alternative to having to worry about passing strings to StringIO, does it make sense to you?

Reply:
Yes-ish. I think it will work, but:

- It may be surprising that constructing the object accepts either str or bytes, but its `.write()` method only accepts bytes.
- The naming is not ideal from a Py3 perspective - I expect a StringIO object to be accepting and returning str data, not bytes. This is my fault from the porting.
- In the longer term, I'd see it as a stopgap solution. It's best for application code to be clear about where it's handling unicode and where it's handling bytes, and minimise the code paths that can be either.